client/js/lounge.js

```diff
@@ -742,13 +757,29 @@ $(function() {
 		}
 	}
 
-	sidebar.on("click", ".chan, button", function() {
+	sidebar.on("click", ".chan, button", function(e, data) {
 		var self = $(this);
 		var target = self.data("target");
 		if (!target) {
 			return;
 		}
 
+		if (!data || data.pushState !== false) {
```

Comment:
I remember now. We want to always default to pushing states unless explicitly told not to (through setting `pushState: false` in the data object). However, since the data variable is arbitrary and may contain other data in the future we need an explicit comparison here. E.g.;

```js
lastChannelJoined.trigger("click", {
    otherData: 'very interesting'
});
```

would not push the new state into history.